### ISSUE TITLE
Can't load cl-unicode system

### ISSUE THREAD

**Issue by ak-coram** [clasp-developers/clasp#1229]
I think the error is caused by the following line:

```
(setq *general-categories* (list '#.(property-symbol "Cn")))
```

The cause seems to be a call to cl-ppcre (e.g. `(canonicalize-name "Cn")`):

```
(defun canonicalize-name (name)
  "Converts the string NAME into a \"canonicalized\" name which can be
used for unambiguous look-ups by removing all whitespace, hyphens, and
underline characters.

Tries not to remove hyphens preceded by spaces if this could lead to
ambiguities as described in
<http://unicode.org/unicode/reports/tr18/#Name_Properties>.

All CL-UNICODE functions which accept string \"names\" for characters
or properties will canonicalize the name first using this function and
will then look up the name case-insensitively."
  (values (ppcre:regex-replace-all "[ _](-A|O-E)$|[-_\\s]" name
                                   (lambda (match register)
                                     (declare (ignore match))
                                     (cond (register (format nil " ~A" register))
                                           (t "")))
                                   :simple-calls t)))
```

It loads just fine with sbcl, but with clasp (current master) I get the following output:

```
Starting cclasp-boehm-0.4.2-4984-g033dc7b77-cst ... loading image...
Top level in: #<PROCESS TOP-LEVEL @0x7fd342c7df09 (Running)>.
COMMON-LISP-USER> (ql:quickload :cl-unicode)
To load "cl-unicode":
  Load 1 ASDF system:
    cl-unicode
; Loading "cl-unicode"
...........; caught ERROR:
;   Read-time evaluation of expression (PROPERTY-SYMBOL
;                                       "Cn") signaled SIMPLE-TYPE-ERROR: In function SUBSEQ, the value of an argument is
;     "Cn"
;   which is not of the expected type (OR NULL UNSIGNED-BYTE)
;     at unknown location
; 

Condition of type: READ-TIME-EVALUATION-ERROR
Read-time evaluation of expression (CL-UNICODE:PROPERTY-SYMBOL #1="Cn") signaled SIMPLE-TYPE-ERROR: In function SUBSEQ, the value of an argument is
  #1#
which is not of the expected type (OR NULL UNSIGNED-BYTE)
Available restarts:
(use :r1 to invoke restart 1, etc.)

1. (RECOVER) Use NIL in place of the missing object.
2. (RETRY) Retry compiling #<CL-SOURCE-FILE "cl-unicode/build" "build" "read">.
3. (ACCEPT) Continue, treating compiling #<CL-SOURCE-FILE "cl-unicode/build" "build" "read"> as having been successful.
4. (RETRY) Retry ASDF operation.
5. (CLEAR-CONFIGURATION-AND-RETRY) Retry ASDF operation after resetting the configuration.
6. (RETRY) Retry ASDF operation.
7. (CLEAR-CONFIGURATION-AND-RETRY) Retry ASDF operation after resetting the configuration.
8. (ABORT) Give up on "cl-unicode"
9. (REGISTER-LOCAL-PROJECTS) Register local projects and try again.
10. (RESTART-TOPLEVEL) Go back to Top-Level REPL.


(ECLECTOR.BASE:%READER-ERROR #<IOSTREAM-STREAM #P"/home/ak/quicklisp/dists/quicklisp/software/cl-unicode-20210228-git/build/read.lisp"> ECLECTOR.READER:READ-TIME-EVALUATION-ERROR :EXPRESSION (CL-UNICODE:PROPERTY-SYMBOL "Cn") :ORIGINAL-CONDITION #<SIMPLE-TYPE-ERROR>)
```

Is this something that would need to be fixed in cl-unicode or cl-ppcre?

**Comment by kpoeck** [comment]
The error is in clasp, see #1183.

Sometimes clasp run cl-ppcre wrong and this error happens.

**Comment by kpoeck** [comment]
Errors in `(property-symbol "Cn")`
which does 
```lisp
(ppcre:regex-replace-all "[ _](-A|O-E)$|[-_\\s]" "Cn"
                                   (lambda (match register)
                                     (declare (ignore match))
                                     (cond (register (format nil " ~A" register))
                                           (t "")))
                                   :simple-calls t)
````

and this errors with calling `(SUBSEQ "Cn" 0 "Cn")``

**Comment by ak-coram** [comment]
I'm now getting a different error with a newer build:

```
$ clasp --load quicklisp/setup.lisp                                <
Starting cclasp-boehm-0.4.2-5011-g5036563fc-cst ... loading image...
Top level in: #<PROCESS TOP-LEVEL @0x7f5ea6feef09 (Running)>.
COMMON-LISP-USER> (ql:quickload :cl-unicode)
To load "cl-unicode":
  Load 1 ASDF system:
    cl-unicode
; Loading "cl-unicode"
.
Condition of type: SIMPLE-ERROR
Unknown property name "OtherMath".
Available restarts:
(use :r1 to invoke restart 1, etc.)

1. (CONTINUE) Retry assertion with new value for CL-UNICODE::DESIGNATOR.
2. (TRY-RECOMPILING) Recompile alias and try loading it again
3. (RETRY) .Retry loading FASL for #<CL-SOURCE-FILE "cl-unicode" "alias">.
4. (ACCEPT) Continue, treating loading FASL for #<CL-SOURCE-FILE "cl-unicode" "alias"> as having been successful.
5. (RETRY) Retry ASDF operation.
6. (CLEAR-CONFIGURATION-AND-RETRY) Retry ASDF operation after resetting the configuration.
7. (RETRY) Retry ASDF operation.
8. (CLEAR-CONFIGURATION-AND-RETRY) Retry ASDF operation after resetting the configuration.
9. (ABORT) Give up on "cl-unicode"
10. (REGISTER-LOCAL-PROJECTS) Register local projects and try again.
11. (RESTART-TOPLEVEL) Go back to Top-Level REPL.


(CORE::ASSERT-FAILURE CL-UNICODE::TEST-FUNCTION (CL-UNICODE::DESIGNATOR) ("OtherMath") "Unknown property name ~S." "OtherMath")
COMMON-LISP-USER>>
```

Any ideas?

**Comment by kpoeck** [comment]
Yes, I also have that, this is miscompiling cl-ppcre.

as a workaround, before cl-ppcre and cl-unicode is compiled, try
`(declaim (optimize (safety 3)(debug 3)(speed 0)))`

Worked just fine for me, but did only tried once

**Comment by kpoeck** [comment]
now tried 5 times:
`(declaim (optimize (safety 3)(debug 3)))`works but
`(declaim (optimize (safety 3)))` does not

**Comment by kpoeck** [comment]
`(declaim (optimize (debug 3)))` is sufficient

**Comment by kpoeck** [comment]
With latest fixes from Bike this now works fine for me, even w/o (declaim (optimize (debug 3))). This was caused by cl-ppcre not being compiled correctly.

So works now fine for me, please retest

**Comment by Bike** [comment]
that's #1183 and my fix 9c216a5 to be clear

**Comment by kpoeck** [comment]
fixed
